File: lesson0/guessthenumber/python/guessthenumber.py
from random import randrange


def init_secret(min, max):
    print(f"Try to guess a number between {minValue} and {maxValue}.")
    return randrange(min, max + 1)


def pick_number():
    while True:
        try:
            guess = int(input("Pick a number... "))
            return guess
        except:
            print("Input valid number, please.")


minValue = 1
maxValue = 100
secret = init_secret(minValue, maxValue)
while True:
    guess = pick_number()
    if guess == secret:
        if input("Yay! You guessed it! Try again? [y/n]: ").lower() != "y":
            break
        secret = init_secret(minValue, maxValue)
    else:
        print(f"The secret is {'greater' if guess < secret else 'less'} than {guess}")

File: lesson0/guessthenumber/csharp/guessthenumber.cs
static int InitSecret(int min, int max)
{
    var random = new Random();
    System.Console.WriteLine($"Try to guess a number between {min} and {max}.");
    return random.Next(min, max);
}

static int PickNumber()
{
    int guess;
    Console.WriteLine("Pick a number... ");
    while (!int.TryParse(Console.ReadLine(), out guess))
    {
        Console.WriteLine("Input valid number, please.");
    }
    return guess;
}

const int minValue = 1;
const int maxValue = 100;
var secret = InitSecret(minValue, maxValue);
while (true)
{
    var guess = PickNumber();
    if (guess == secret)
    {
        Console.Write("Yay! You guessed it! Try again? [y/n]:");
        if (Console.ReadLine().ToLower() != "y")
        {
            break;
        }
        secret = InitSecret(minValue, maxValue);
    }
    else
    {
        Console.WriteLine($"The secret is {(guess < secret ? "greater" : "less")} than {guess}");
    }
}

